modules/media.js

```diff
@@ -157,15 +183,41 @@ Media.prototype.restore = function(type, oncomplete) {
         type: mimeType
       });
 
-      ffosbr.media.put(type === 'photos' ? 'pictures' : type, newFile, filename, oncomplete);
+      ffosbr.media.put(type === 'photos' ? 'pictures' : type, newFile, filename, function(error) {
+        if (error) {
+          // If the put fails, break the callback chain. The restore has failed.
+          oncomplete(type, error);
+        } else {
+          // If not, recurse on remaining files to be written.
+          writeFiles(remaining);
+        }
+
+        // TODO - Report progress?
+
+      });
     };
 
     reader.readAsArrayBuffer(file);
-  }, oncomplete);
+  };
+
+  // Ignore any errors in "get". They will show up in the oncomplete.
+  ffosbr.media.get('sdcard1', paths[type], function(file, error) {
+    if (!file) {
+      empty = true;
```

Comment:
This is broken; `empty` is initialized to `true` and never set to `false`.

Also, since `media.get()` checks to see whether `file` exists before entering `forEach`, this branch is never taken.